gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt

```diff
@@ -143,37 +143,37 @@ internal class AvatarPickerViewModel(
                     failedUploadDialog = null,
                 )
             }
-            when (val result = avatarRepository.uploadAvatar(email, uri)) {
+            val selectAvatar = _uiState.value.emailAvatars?.selectedAvatarId == null
+            when (val result = avatarRepository.uploadAvatar(email, uri, selectAvatar)) {
                 is GravatarResult.Success -> {
                     fileUtils.deleteFile(uri)
-                    val isAutoSelected = _uiState.value.emailAvatars?.selectedAvatarId == null
-                    if (isAutoSelected) {
-                        fetchAvatars(showLoading = false, scrollToSelected = true)
```

Comment:
We no longer need this. The Avatar in the response will now have a proper `selected` value set so we can update the UI based on that.